### ISSUE TITLE
Grid - Rendering custom components in filter header area based on user selected FilterOperation

### ISSUE THREAD

**Issue by TWhidden** [DevExpress/devextreme-reactive#3059]
- [x ] I have searched this repository's [issues](https://github.com/devexpress/devextreme-reactive/issues) and believe that this is not a duplicate.

### I'm using ...

- [ x] React Grid
- [ ] React Chart
- [ ] React Scheduler

### Description

I am looking to render my own control based on the filter selection.  Looking through the GitHub source as well as the docs, I cant seem to find where I can gain access to `FilterOperation` when its changed, while creating my own editor component.

![image](https://user-images.githubusercontent.com/9921122/91252525-9bf25900-e712-11ea-8f89-36106db64112.png)

![image](https://user-images.githubusercontent.com/9921122/91252568-ab71a200-e712-11ea-8c4e-4d2a7ee93372.png)

Obviously, `DataTypeProvider.ValueEditorProps` does not contain a `filterOperation`, and unsure if it should or not.  But this editor appears to be responsible for rendering the user input filter area each time I select the different options in the drop down filter area.

![image](https://user-images.githubusercontent.com/9921122/91253077-d4466700-e713-11ea-8126-dfabd5dd9fea.png)

![image](https://user-images.githubusercontent.com/9921122/91252735-01dee080-e713-11ea-8c82-28e442711d1f.png)

At the time of returning this control, I could return components related to the user selected drop down, that acquire data in different ways.

Can you tell me the proper way to capture the user selected filter option from my DataTypeProvider, so I can render that component?

### Environment
- devextreme-reactive: 2.7.0
- react: 16.13.1
- browser: any
- material-ui: yes

**Comment by LazyLahtak** [comment]
Hi @TWhidden,

We have a demo for this scenario on our site. Please refer to it: [React Grid - Filtering. Custom Filter Operations](https://devexpress.github.io/devextreme-reactive/react/grid/docs/guides/filtering/#custom-filter-operations).
If you have any further questions, let us know.

**Comment by TWhidden** [comment]
This @LazyLahtak  - I did review those docs but it wasn't really clear to me how to capture that `FilterOperation` so I could change the filter component based on the selected operation. There was another request in the past issues that did help me through this.  In short, and reference to others if it comes up, I ended up just capturing the `FilterOperation` as shown in this example:

https://codesandbox.io/s/devextreme-react-grid-for-material-ui-jxrgn?file=/index.js

Which got me what I needed. Thanks for following up.

**Comment by LazyLahtak** [comment]
Yes, the `TableFilterRow.FilterSelector` component doesn't get the column name, but on lines 164-166 we provide additional properties to the `TableFilterRow.Cell` component's children. These children are `TableFilterRow.FilterSelector` and `TableFilterRow.Editor`. Also, we don't pass redundant properties to the component. For this reason, these components don't get the `columnName` property by default.

The `DataTypeProvider` plugin uses `availableFilterOperations` to change the list of filters for necessary columns. These filters (editors and selectors) are rendered by the `TableFilterRow` plugin. `DataTypeProvider` allows you to change rendered editors for necessary columns (all editors, not only for the filter row).

Note that the `DataTypeProvider.ValueEditor` component also gets additional properties from the replaced editor. So, I suggest that you pass additional properties to the `DataTypeProvider.ValueEditor` component in the same way as we passed them to `TableFilterRow.FilterSelector` and `TableFilterRow.Editor`. Here is an [example](https://codesandbox.io/s/devextreme-react-grid-for-material-ui-forked-wyjzq).

Please check it and let me know if it meets your requirements.

**Comment by TWhidden** [comment]
Thanks for the help with this.  I was able to get it to work following the examples. Just using TypeScript trying to stay within those type constraints and not using the `any` type as much as I can.

In short, FilterSelector, capture the filter operation for the column being processed _(Line 135 index.js)_
`DataTypeProvider.editorComponent`, pull from state the current operation for the column being processed, and render desired component. _(Line 126, index.js)_

Im am not claiming I did this right yet, just that I did accomplish the goal. I'll continue to study the Grid.  Thanks for the help.
